### PR TITLE
Add HyperScript syntax

### PR DESCRIPTION
**Pull request by roman01la**
Using `$` macro to create React elements instead of `#el` tag

### CODE REVIEW

**Comment by accounts-pitch-io** [comment]
master:
React 309ms
UIx 514ms 1.7x
Reagent 834ms 2.7x


PR:
React 431ms
UIx 561ms 1.3x
Reagent 1022ms 2.4x

**Comment by accounts-pitch-io** [comment]
master:
React 309ms
UIx 514ms 1.7x
Reagent 834ms 2.7x


PR:
React 369ms
UIx 543ms 1.5x
Reagent 979ms 2.7x

**Comment by accounts-pitch-io** [comment]
master:
React 309ms
UIx 514ms 1.7x
Reagent 834ms 2.7x


PR:
React 341ms
UIx 613ms 1.8x
Reagent 1037ms 3x

**Comment by accounts-pitch-io** [comment]
master:
React 309ms
UIx 514ms 1.7x
Reagent 834ms 2.7x


PR:
React 325ms
UIx 508ms 1.6x
Reagent 858ms 2.6x

File: core/test/uix/core_test.clj [hunk 1]
```diff
@@ -20,3 +20,11 @@
                         (uix.core/parse-sig 'component-name '(([props]) ([props x])))))
   (is (thrown-with-msg? AssertionError #"uix.core\/defui should be a single-arity component"
         (uix.core/parse-sig 'component-name '([props x])))))
+
+(deftest test-h
+  (is (= (macroexpand-1 '(uix.core/$ :h1))
+         '(uix.compiler.aot/>el "h1" (cljs.core/array nil) (cljs.core/array))))
+  (is (= (macroexpand-1 '(uix.core/$ :> identity {} 1 2))
+         '(uix.compiler.aot/>el identity (cljs.core/array (cljs.core/js-obj)) (cljs.core/array 1 2))))
+  (is (= (macroexpand-1 '(uix.core/$ :> identity {:x 1 :ref 2} 1 2))
```
Comment (codebeige) [inline core/test/uix/core_test.clj:29]:
```suggestion
  (is (= (macroexpand-1 '(uix/$ :h1))
         '(uix.compiler.aot/>el "h1" (cljs.core/array nil) (cljs.core/array))))
  (is (= (macroexpand-1 '(uix/$ :> identity {} 1 2))
         '(uix.compiler.aot/>el identity (cljs.core/array (cljs.core/js-obj)) (cljs.core/array 1 2))))
  (is (= (macroexpand-1 '(uix/$ :> identity {:x 1 :ref 2} 1 2))
```

File: core/test/uix/core_test.cljs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 (ns uix.core-test
   (:require [clojure.test :refer [deftest is async testing run-tests]]
-            [uix.core :as uix.core :refer [defui]]
+            [uix.core :as uix.core :refer [defui $]]
```
[the remark this answers is not in the document]
Comment (codebeige) [reply core/test/uix/core_test.cljs:3]: ... and then replace `uix.core/defui` and friends with `uix/defui` across the file.

**Review by codebeige — APPROVED** [review]
Looks good! 🎉

**Comment by accounts-pitch-io** [comment]
master:
React 309ms
UIx 514ms 1.7x
Reagent 834ms 2.7x


PR:
React 315ms
UIx 519ms 1.6x
Reagent 841ms 2.7x

**Review by dundalek — APPROVED** [review]
Awesome :rocket:

File: core/src/uix/core.clj [hunk 3]
```diff
@@ -56,3 +56,13 @@
   "Returns source string of UIx component"
   [sym]
   (uix.source/source sym))
+
+(defmacro $
+  "Creates React element
+
+  DOM element: ($ :button#id.class {:on-click handle-click} \"click me\")
```
[the remark this answers is not in the document]
Comment (roman01la) [reply core/src/uix/core.clj:63]: Good point, RUM is doing that as well. I'll add this in a follow up PR

**Comment by accounts-pitch-io** [comment]
master:
React 309ms
UIx 514ms 1.7x
Reagent 834ms 2.7x


PR:
React 377ms
UIx 606ms 1.6x
Reagent 1015ms 2.7x

**Pull request merged**